iib/workers/tasks/build.py

```diff
@@ -116,48 +115,36 @@ def _create_and_push_manifest_list(request_id, arches):
     :rtype: str
     :raises IIBError: if creating or pushing the manifest list fails
     """
+    buildah_manifest_cmd = ['buildah', 'manifest']
     output_pull_spec = get_rebuilt_image_pull_spec(request_id)
-    log.info('Creating the manifest list %s', output_pull_spec)
-    with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
-        manifest_yaml = os.path.abspath(os.path.join(temp_dir, 'manifest.yaml'))
-        with open(manifest_yaml, 'w+') as manifest_yaml_f:
-            manifest_yaml_f.write(
-                textwrap.dedent(
-                    f'''\
-                    image: {output_pull_spec}
-                    manifests:
-                    '''
-                )
-            )
-            for arch in sorted(arches):
-                arch_pull_spec = _get_external_arch_pull_spec(request_id, arch)
-                log.debug(
-                    'Adding the manifest %s to the manifest list %s',
-                    arch_pull_spec,
-                    output_pull_spec,
-                )
-                manifest_yaml_f.write(
-                    textwrap.dedent(
-                        f'''\
-                        - image: {arch_pull_spec}
-                          platform:
-                            architecture: {arch}
-                            os: linux
-                        '''
-                    )
-                )
-            # Return back to the beginning of the file to output it to the logs
-            manifest_yaml_f.seek(0)
-            log.debug(
-                'Created the manifest configuration with the following content:\n%s',
-                manifest_yaml_f.read(),
+    log.info('Creating the manifest list %s locally', output_pull_spec)
+    sorted_arches = sorted(arches)
+    for arch in sorted_arches:
+        arch_pull_spec = _get_external_arch_pull_spec(request_id, arch, include_transport=True)
+        log.debug(
+            'Adding the manifest %s to the manifest list %s', arch_pull_spec, output_pull_spec
+        )
+        # Create the manifest list with the first arch and then add other arches to it using the
+        # buildah-manifest add command.
+        if arch == sorted_arches[0]:
+            run_cmd(
+                buildah_manifest_cmd + ['create', output_pull_spec, arch_pull_spec],
```

Comment:
You can call `create` with just the pull spec of the manifest list. Consider this pattern:
```python
run_cmd(
    buildah_manifest_cmd + ['create', output_pull_spec],
    exc_msg=f'Failed to create the manifest list locally: {output_pull_spec}',
)
...
for arch in sorted_arches:
...
  run_cmd(
       buildah_manifest_cmd + ['add', output_pull_spec, arch_pull_spec],
       exc_msg=(
          f'Failed to add {arch_pull_spec} to the' f' local manifest list: {output_pull_spec}'
       ),
  )
...
```

It has the benefit of not treating the first arch a special case.

Reply:
Done